packages/edit-navigation/src/index.js

```diff
@@ -43,12 +43,30 @@ const fetchLinkSuggestions = async ( search, { perPage = 20 } = {} ) => {
 		} ),
 	} );
 
-	return map( posts, ( post ) => ( {
-		id: post.id,
-		url: post.url,
-		title: decodeEntities( post.title ) || __( '(no title)' ),
-		type: post.subtype || post.type,
-	} ) );
+	const categories = await apiFetch( {
+		path: addQueryArgs( '/wp/v2/search', {
+			search,
+			per_page: perPage,
+			type: 'category',
+		} ),
+	} );
+
+	const formats = await apiFetch( {
+		path: addQueryArgs( '/wp/v2/search', {
+			search,
+			per_page: perPage,
+			type: 'post-format',
+		} ),
+	} );
```

Comment:
The PR might be taking a different approach, but I noticed the `await`s need to be removed here to make the `Promise.all` work.